streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1225,14 +1226,15 @@ boolean usesPatternSubscription() {
     }
 
     synchronized Pattern sourceTopicPattern() {
-        log.debug("Found pattern subscribed source topics, falling back to pattern subscription for the main consumer.");
-
         if (topicPattern == null) {
             final List<String> allSourceTopics = maybeDecorateInternalSourceTopics(sourceTopicNames);
             Collections.sort(allSourceTopics);
             topicPattern = buildPattern(allSourceTopics, nodeToSourcePatterns.values());
         }
 
+        log.debug("Found pattern subscribed source topics, falling back to pattern " +
+            "subscription for the main consumer: {}", topicPattern);
+
```

Comment:
Not sure if we intended to move this log message. Also not sure if we really need this log message at all.

Reply:
I raised this to @ableegoldman in another PR about this already, we will do some further cleanup after this PR (but it was not introduced by this JIRA, I think it was there long time ago: https://github.com/apache/kafka/commit/57b2f6807d332f4e6de4dc4ad18aa24686d304de#r37098609